crates/modelardb_server/src/storage/compressed_data_manager.rs

```diff
@@ -113,55 +119,97 @@ impl CompressedDataManager {
         StorageEngine::write_batch_to_apache_parquet_file(record_batch, file_path.as_path(), None)
     }
 
+    /// Read and process messages received from the [`super::UncompressedDataManager`] to either
+    /// ingest compressed data, flush buffers, or stop.
+    pub(super) fn process_compressed_messages(&self, runtime: Arc<Runtime>) -> Result<(), String> {
+        loop {
+            let message = self
+                .channels
+                .compressed_data_receiver
+                .recv()
+                .map_err(|error| error.to_string())?;
+
+            match message {
+                Message::Data(compressed_segment_batch) => {
+                    runtime
+                        .block_on(self.insert_compressed_segments(compressed_segment_batch))
+                        .map_err(|error| error.to_string())?;
+                }
+                Message::Flush => {
+                    runtime.block_on(self.flush()).unwrap();
+                    self.channels
+                        .result_sender
+                        .send(Ok(()))
+                        .map_err(|error| error.to_string())?;
+                }
+                Message::Stop => {
+                    runtime.block_on(self.flush()).unwrap();
```

Comment:
Maybe add a comment to explain why unwrap is safe or map the error as above.

Reply:
Fixed.